packages/react-router-dev/vite/plugin.ts

```diff
@@ -1244,6 +1245,20 @@ export const reactRouterVitePlugin: ReactRouterVitePlugin = (_config) => {
         }
       },
     },
+    {
+      name: "react-router-define-route",
+      enforce: "pre",
+      async transform(code, id, options) {
+        if (id.endsWith(BUILD_CLIENT_ROUTE_QUERY_STRING)) return;
+        if (!code.includes("defineRoute")) return; // temporary back compat
+
+        if (options?.ssr) return; // TODO: why is `ctx` undefined otherwise?
```

Comment:
@markdalgleish I guess we never ran into this before just out of luck, but seems like `ctx` is `undefined` when `ssr` is set to `true` which causes a bunch of issues. Seems surprising to me, so might be missing something